modules/core/src/main/java/org/apache/ignite/internal/util/IgniteUtils.java

```diff
@@ -2223,6 +2226,19 @@ public static synchronized boolean isLocalHostChanged() throws IOException {
                     }
                 }
 
+                /* On z/OS NetworkInterface will expose IPv6 ::1 and IPv4 external address, but not 127.0.0.1
+                    "localhost" resolves to 127.0.0.1 so we need to add it to binding addresses list so
+                    that discovery and clients can connect to "localhost" ports. */
```

Comment:
I'd reword the comment like this based on our conversation:

```suggestion
                /* On z/OS NetworkInterface::getNetworkInterfaces returns external interfaces 
                    and IPv6 loopback but not IPv4 loopback.
                    Trying to additionally resolve "localhost" to find the IPv4 loopback address. */
```